src/services/download-service.ts

```diff
@@ -12,123 +13,153 @@ class DownloadService {
+        {loader: FlavorsLoader, params: {entryId}},
+        {loader: AttachmentsLoader, params: {entryId}}
+      ],
+      ks
+    );
 
-      if (isContentTypeSupported && fileName) {
-        return {
-          downloadUrl,
-          fileName
-        };
+    return this.parseDataFromResponse(data);
+  }
+
+  private parseDataFromResponse(data: Map<string, any>): object {
+    const kalturaAssets = {
+      captions: [],
+      flavors: [],
+      attachments: []
+    };
+
+    if (data) {
+      if (data.has(CaptionsLoader.id)) {
+        const captionsLoader = data.get(CaptionsLoader.id);
+        kalturaAssets.captions = captionsLoader?.response?.captions;
       }
-    } catch (e: any) {
-      this.logger.warn('Failed to get file from url: ', requestUrl);
-      // in case HEAD request failed for raw service (image use-case), retry to get the image download metadata using thumbnail service
-      if (this.player.isImage()) {
-        return await this.getImageDownloadMetadata();
+      if (data.has(FlavorsLoader.id)) {
+        const flavorsLoader = data.get(FlavorsLoader.id);
+        kalturaAssets.flavors = flavorsLoader?.response?.flavors;
+      }
+      if (data.has(AttachmentsLoader.id)) {
+        const attachmentsLoader = data.get(AttachmentsLoader.id);
+        kalturaAssets.attachments = attachmentsLoader?.response?.attachments;
       }
     }
 
-    return null;
+    return kalturaAssets;
   }
 
-  private async getImageDownloadMetadata(): Promise<DownloadMetadata> {
+  private async getImageDownloadMetadata(): Promise<string> {
     // eslint-disable-next-line @typescript-eslint/ban-ts-comment
     // @ts-ignore
     const imageSource = this.player.config.sources.image;
     let requestUrl = imageSource && imageSource.length > 0 ? imageSource[0].url : '';
-    if (!requestUrl) return null;
+    if (!requestUrl) return '';
     if (!requestUrl.includes('/ks/')) {
       const ks = this.player.config.session.ks;
       requestUrl = ks ? `${requestUrl}/ks/${ks}` : requestUrl;
     }
     try {
       const response = await fetch(requestUrl);
       const blobImage = await response.blob();
-      const href = URL.createObjectURL(blobImage);
-      return {
-        downloadUrl: href,
-        fileName: this.player.sources?.metadata?.name || 'image'
-      };
+      return URL.createObjectURL(blobImage);
     } catch (e: any) {
       this.logger.warn('Failed to get image from url: ', requestUrl);
     }
-    return null;
+    return '';
   }
 
-  private getRequestUrl(config: DownloadConfig): string {
-    if (this.player.isImage()) {
-      const requestUrl = this.player.sources.downloadUrl;
-      if (!requestUrl) return '';
-      const ks = this.player.config.session.ks;
-      return ks ? `${requestUrl}/ks/${ks}` : requestUrl;
+  private async _getDownloadUrl(downloadUrl: string): Promise<string> {
+    if (downloadUrl.includes('caption_captionAsset')) {
+      try {
+        const response = await fetch(downloadUrl);
```

Comment:
note that if any of the async calls fail, the code will fall into the catch block, so make sure that all the information about the error that you need would be printed there. do you want to have the same message for when the fetch fails and for when the blob creation fails ? (yes is a legitimate answer)